qualtran/bloqs/data_loading/one_hot_encoding_test.py

```diff
@@ -0,0 +1,57 @@
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+import attrs
+import cirq
+import pytest
+from attr import field
+from numpy._typing import NDArray
+
+from qualtran import GateWithRegisters, QUInt, Signature
+from qualtran.bloqs.data_loading.one_hot_encoding import OneHotEncoding
+from qualtran.cirq_interop.bit_tools import iter_bits
+from qualtran.cirq_interop.testing import assert_circuit_inp_out_cirqsim
+
+
+@attrs.frozen
+class OneHotEncodingTest(GateWithRegisters):
+    integer: int = field()
+    size: int = field()
+
+    @property
+    def signature(self) -> 'Signature':
+        return Signature.build_from_dtypes(a=QUInt(self.size), b=QUInt(2**self.size))
+
+    def decompose_from_registers(
+        self, *, context: cirq.DecompositionContext, **quregs: NDArray[cirq.Qid]
+    ) -> cirq.OP_TREE:
+        a = quregs['a']
```

Comment:
I would be careful here with the endianness. I think you should be reversing the qubit order here and not in your test (iter_bits). Qualtran assumes a big endian convention.

Reply:
Done